src/components/BaseTimer.js

```diff
@@ -0,0 +1,147 @@
+import React, { Component } from 'react';
+import '../App.css';
+
+const MAX_DURATION = 5940000; // max duration 99 minutes
+const ONE_SECOND = 1000;
+const ONE_MINUTE = 60 * ONE_SECOND;
+
+class BaseTimer extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      timerOn: false,
+      countDown: true, // probably can go in redux to be shared in other Timers
+      setTime: 0,
+      currentTime: 0,
+    };
+  }
+
+  getTimerAdjustButtons() {
+    return (
+      <div className="BaseTimer-display">
+        <button type="button" onClick={() => this.adjustTimer('incMinutes')}>&#8679;</button>
+        <button type="button" onClick={() => this.adjustTimer('decMinutes')}>&#8681;</button>
+        <button type="button" onClick={() => this.adjustTimer('incSeconds')}>&#8679;</button>
+        <button type="button" onClick={() => this.adjustTimer('decSeconds')}>&#8681;</button>
+      </div>
+    );
+  }
+
+  startTimer = () => {
+    const { countDown, currentTime, setTime } = this.state;
+    const newSetTime = setTime || currentTime;
+    let currentCountUpTime;
+
+    if (!countDown) {
+      currentCountUpTime = setTime ? currentTime : 0;
+      // currentCountUpTime logic -> if starting for 1st time, currentTime must be 0.
+      // When starting from pause, should be currentTime
+      // may still need some kind of flag for set but paused,
+      // but that should be capturable from setTime
+      // can also consider another button for clearing everything
+    }
+
+    this.setState({
+      timerOn: true,
+      setTime: newSetTime,
+      // Always want the set time to be what the clock was orignally set for.
+      // Think about how not to overwrite this when paused
+      currentTime: currentCountUpTime != null ? currentCountUpTime : currentTime,
+    });
+
+    this.timer = setInterval(() => {
+      let newTime;
+      if (countDown) {
+        newTime = this.state.currentTime - 10;
+      } else {
+        newTime = this.state.currentTime + 10;
```

Comment:
`currentTime` was set via object destructuring assignment on state in the outer function, but since we're changing the value of `this.state.currentTime` below in `setInterval()`, we have to reference `this.state.currentTime` here and not the static value stored in `currentTime`

This is also true below on line 61 with `setTime`